### DISCUSSION TITLE
AG ramp up issue : unable to load a session in DAW

### DISCUSSION THREAD

**Discussion by Henri75** [apohl79/audiogridder#1075] [Help]
I already noticed that issue, but right now I am travelling with only a macbook pro i9 (normally I have a M1 max) and I cant load most sessions because almost all plugins stay red or orange when loading a local session.
It seems to be related to ramp up : when loading the sessions all ag instances are reloaded at almost the same time, creating huge cpu spikes > 90% usage and then AG instances dont load.
If I kill the local AG server  and then instantiate or load each instance one after the other, then everything will load and my cpu usage after everything is loaded is 19% ! So this computer should be powerfull enough to load the session with the local AG server and instances.
But I can't  recall the session directly from the DAW because of the cpu spikes when loading the 15+ AG instances.
I guess some mecanism of ramp up (for example incremental loading or short pause between loading of each AG instances, with a limit of concurrent instances loaded) will help solve that issue !

**Comment by masopmac** [comment]
First of all this is incredible software. I wish it was built-into the DAW. Similar issue here but I'm pushing my hardware pretty close to max 80% + CPU when all plugs are running. At idle it's about 34% usage.  

Running ~120 mostly Acustica plugins w/ some UAD as well per session. Killing the local AG Server and reloading in my case doesn't seem to load everything, neither does "Reload all "not loaded" chains" from the "AGridder Monitor". 

What does seem to work:

Once processing settles, after initial project open, with about 50% or more of total plugin channels green (ok) and the remaining orange (not loaded). Right-clicking on the "AGridder Monitor", under the "Connections" menu finding the unloaded "Channel" (by name) on one of the local server instances and selecting a different local sever (by ID) will usually flip the plugin straight to green from orange. Another thing that works is just finding the not loaded "Agridder" plugin on the affected DAW channel and clicking the little server icon and "Reload". 100% of the time this will load all of the remaining not loaded plugins. 

I've been thinking a feature similar to "Reload all "not loaded" chains"  but does a slower one-by-one more synchronous and delayed loading attempt (maybe the delay between loading channels is configurable?) would be cool to catch the not loaded channels and eliminate some of the extra clicking to get things to run. 

Config: 

macOS 12.6 
AudioGridder 1.2.0-beta15 - Local Mode
Logic Pro 10.7.4
64GB of RAM w/ fast Intel Chip
